### DISCUSSION TITLE
How to validate/dry-run template file

### DISCUSSION THREAD

**Discussion by tangkhaiphuong** [hairyhenderson/gomplate#2210] [Q&A]
Hi, I'm using gomplate to render template file with use many environment variables. The process is failed if missing any env-var. It's inconvenient.

Is there any way/option/configure that can help validate/dry-run template and list out what env-var are missing? Similar with some compiler can show list of error instead of only one. It's help save a lot of time for development. 

For example. I have my template file `test.tpl`:

```
First Name: {{ .Env.FIRST_NAME }}
Last Name: {{ .Env.LAST_NAME }}
```

Then I try to render
```
gomplate -f test.tpl
First Name: 15:13:26 ERR  err="renderTemplate: failed to render template test.tpl: template: test.tpl:1:19: executing \"test.tpl\" at <.Env.FIRST_NAME>: map has no entry for key \"FIRST_NAME\""
```

If support some think:
```
gomplate check -f test.tpl

Failed to render template test.tpl: template: test.tpl:1:19: executing \"test.tpl\" at 
  <.Env.FIRST_NAME>: map has no entry for key \"FIRST_NAME\""
  <.Env.LAST_NAME>: map has no entry for key \"LAST_NAME\""
  ...
```

**Comment by tewfik-ghariani** [comment]
You can use `getenv` instead which does not fail if the environment variable is not set ^^
```
 $ cat test.tpl
First Name: {{ getenv "FIRST_NAME" }}
Last Name: {{ getenv "LAST_NAME" }}
```
```
 $ echo $FIRST_NAME
tang
 $ echo $LAST_NAME

 $ gomplate -f test.tpl
First Name: tang
Last Name:
```
